src/app/components/pages/MyProgress.tsx

```diff
@@ -37,6 +39,41 @@ const statistics = {
   tagColWidth: "col-lg-12",
 };
 
+const QuestionParts = ({ progress }: { progress: MyProgressState | undefined }) => (
+  <div className="mt-4">
+    <h4>Question parts correct by Type</h4>
+    <Row>
+      {statistics.questionTypeStatsList.map((qType: string) => {
+        const correct = progress?.correctByType?.[qType] ?? null;
+        const attempts = progress?.attemptsByType?.[qType] ?? null;
+        const percentage = safePercentage(correct, attempts);
+        return (
+          <Col key={qType} className={`${statistics.typeColWidth} mt-2 type-progress-bar`}>
+            <div className={"px-2"}>{HUMAN_QUESTION_TYPES[qType]} questions correct</div>
+            <div className={"px-2"}>
+              <ProgressBar percentage={percentage ?? 0}>
+                {percentage == null ? "No data" : `${correct} of ${attempts}`}
+              </ProgressBar>
+            </div>
+          </Col>
+        );
+      })}
+    </Row>
+  </div>
+);
+
+const getPageTitle = ({ userDetails, viewingOwnData }: { userDetails?: UserSummaryDTO; viewingOwnData: boolean }) => {
```

Comment:
Defining a single parameter as an object and then destructuring like you're doing here makes sense when you call this function and pass an object that already contains all the parameters as its attributes. In your code this is not the case, you're manually creating an object and setting its attributes individually, so it makes more sense to define this as:

```typescript
const getPageTitle = (viewingOwnData: boolean, userDetails?: UserSummaryDTO) => {
```

The same applies to `getTagData`.